dev-docs/git-practices.md

```diff
@@ -0,0 +1,133 @@
+# Forks and PR workflows
+
+Changes to the Hail codebase are made through pull requests directly to the
+`main` branch of the `hail-is/hail` repository. Here are the steps to
+take when contributing for the first time and when making a pull request.
+
+
+## First time actions
+
+In order to keep the `hail-is/hail` repo clean, we ask that contributors develop
+features in branches on a personal fork. The first step then is to [fork
+the repository](https://github.com/hail-is/hail/fork).
+
+Once you've done so, clone the repository from your fork. This will set up a
+local copy of your fork with a single remote called `origin`. In this document,
+we will refer to `origin` as your developer fork, and `upstream` as `hail-is/hail`.
+You can check which origins you have configured by running `git remote -v`.
+
+While feature branches will live on your fork, those branches are still going to want
+to be based on the latest changes in `hail-is/hail:main`. So we will add the
+upstream repository as another remote so we can pull in those changes.
+
+```bash
+git remote add upstream https://github.com/hail-is/hail.git
+```
+
+If you run `git remote -v` again, you should see something like the following:
+
+```
+origin	https://github.com/<your-github-username>/hail.git (fetch)
+origin	https://github.com/<your-github-username>/hail.git (push)
+upstream	https://github.com/hail-is/hail.git (fetch)
+upstream	https://github.com/hail-is/hail.git (push)
+```
+
+To get the latest changes on your local `main` branch, run
```

Comment:
Why even have this? I find it easier to just always refer to `hi/main` and it always means what I want it to mean.

Reply:
ah, good point. I'll change to that

Reply:
Oh yeah, that's preferable. I usually do `g checkout hi/main ; g checkout -b feature` but yours is better.